src/components/Grocery.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react'
+import { ACTIONS } from '../App.jsx'
+
+export default function Grocery({ grocery, dispatch }) {
+  const [editedGrocery, setEditedGrocery] = useState('');
+
+  function handleEdit(e) {
+    e.preventDefault();
+    dispatch({ type: ACTIONS.UPDATE, payload: { id: grocery.id, name: editedGrocery } });
+    setEditedGrocery('');
+  }
+  return (
+    <div>
+      {grocery.editing ?
+        <form onSubmit={handleEdit}>
+          <input 
+            type='text'
+            value={editedGrocery}
+            onChange={(e) => setEditedGrocery(e.target.value)}
+          />
+          <button>
+            Save
+          </button>
+        </form> :
+        <>
+          <span
+            style={{ color: grocery.inCart ? '#aaa' : '#000' }}
+            onClick={() => dispatch(
+              { type: ACTIONS.PUT_IN_CART, payload: { grocery } }
+            )}
+          >
+            {grocery.name}
+          </span>
+          <button
+            onClick={() => dispatch(
```

Comment:
When I am editing an item, the item name doesn't state in the input box so I am unsure what I am edit. When I leave it blank it updates to be blank.